File: calculator.go
package main
import "fmt" // needed Println and Scanln

func main(){
  fmt.Println("***************************")
  fmt.Println(" A SIMPLE CALCULATOR IN GO ");
  fmt.Println("ENTER INPUTS WITHOUT SPACES")
  fmt.Println("***************************\n\n")

  for true {
    var option rune; // rune == char
    var line string;
    fmt.Print("Enter an operation (+, -, *, /) | 'q' to quit: "); 
    _,_= fmt.Scanln(&line); // get input but ignore both returns
    option = rune(line[0]); // get the first rune of the input
    if option == 'q' || option == 'Q'{ break }
    var num1, num2 float64; // (8-byte float == double)
    fmt.Print("Enter number 1 (double): ");
    _,_ = fmt.Scan(&num1); // dont care about errors or number of items 
    fmt.Print("Enter number 2 (double): ");
    _,_ = fmt.Scan(&num2); // dont care about errors or number of items 
    total := 0.0;
    
    switch option{
    case '+':
      total = num1 + num2;
    case '-':
      total = num1 - num2;
    case '*':
      total = num1 * num2;
    case '/':
      total = num1 / num2;
  }
    fmt.Println("The result is:", total);
  }
}

File: calculator.cpp
#include <iostream> // needed for std::cout and std::cin

using std::cout, std::cin; // better than doing "using namespace std;"
int main(){
  cout << "**************************\n";
  cout << "A SIMPLE CALCULATOR IN C++\n";
  cout << "ENTER INPUT WITHOUT SPACES\n";
  cout << "**************************\n\n\n";

  while(true){
    cout << "Enter an operation (+, -, *, /) | 'q' to quit: ";
    char option;
    cin >> option; // get char input
    if(option == 'q' || option == 'Q') break;
    double num1, num2, total;
    cout << "Enter number 1 (double): ";
    cin >> num1; 
    cout << "Enter number 2 (double): ";
    cin >> num2;

    switch(option){
      case '+': total = num1 + num2;
      break;
      case '-': total = num1 - num2;
      break;
      case '*': total = num1 * num2;
      break;
      case '/': total = num1 / num2;
      break;
    }
    cout << "The result is " << total << "\n";
  }
  return 0;
}
